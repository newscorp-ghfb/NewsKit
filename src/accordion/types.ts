import React from 'react';
import {NewsKitIconProps} from '../icons';
import {MQ} from '../utils';
import {LogicalPaddingProps} from '../utils/logical-properties';
import {Override} from '../utils/overrides';
import {TransitionToken} from '../theme';

export type AccordionIconProps = NewsKitIconProps &
  Pick<AccordionProps, 'expanded'>;

export interface AccordionHeaderOverrides extends LogicalPaddingProps {
  minWidth?: MQ<string>;
  minHeight?: MQ<string>;
  stylePreset?: MQ<string>;
  typographyPreset?: MQ<string>;
  spaceInline?: MQ<string>;
  label?: {
    typographyPreset?: MQ<string>;
  };
  indicatorIcon?: Override<AccordionIconProps>;
  transitionPreset?: TransitionToken | TransitionToken[];
}

export interface AccordionPanelOverrides {
  stylePreset?: MQ<string>;
  transitionPreset?: TransitionToken | TransitionToken[];
}

export type AccordionPropsOverrides = {
  header?: AccordionHeaderOverrides;
  panel?: AccordionPanelOverrides;
};

export interface AccordionProps {
  children?: Exclude<React.ReactNode, 'undefined'>;
  disabled?: boolean;
  header?: Exclude<React.ReactNode, 'undefined'>;
  headerAs?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span';
  label?: string;
  ariaControls?: string;
  id?: string;
  expanded?: boolean;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  overrides?: AccordionPropsOverrides;
  onChange?: (expanded: boolean) => void;
}

export type AccordionGroupProps = {
  defaultExpanded?: number | number[] | 'all';
  expanded?: number | number[] | 'all';
  onChange?: (expanded: number[]) => void;
  expandSingle?: boolean;
  children: React.ReactNode;
} & Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>;
