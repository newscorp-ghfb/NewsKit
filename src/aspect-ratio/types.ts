/* eslint-disable import/no-extraneous-dependencies */
import {
  MaxHeightProperty,
  MaxWidthProperty,
  PaddingTopProperty,
  HeightProperty,
  WidthProperty,
} from 'csstype';
import {GetAspectRatioProp} from '../utils/get-aspect-ratio';

export interface ContainerProp {
  paddingTop?: PaddingTopProperty<string>;
  maxHeight?: MaxHeightProperty<string>;
  maxWidth?: MaxWidthProperty<string>;
}

export interface StyledContainerProps {
  $height?: HeightProperty<string>;
  $width?: WidthProperty<string>;
}

/**
 * @deprecated AspectRatioProps is deprecated and will be removed in the next major release.
 */
export interface AspectRatioProps
  extends GetAspectRatioProp,
    Omit<ContainerProp, 'paddingTop'> {}
