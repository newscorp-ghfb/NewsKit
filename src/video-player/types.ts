import {LogicalProps} from '../utils/logical-properties';
import {MQ} from '../utils/style';

declare global {
  interface Window {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    videojs?: any;
  }
}

export interface BrightCoveBasicAttributes {
  'data-account': string;
  'data-player': string;
  'data-video-id'?: string;
  'data-playlist-id'?: string;
  'data-embed': string;
  controls?: boolean;
}

export type BrightCoveOtherAttributes = object;

type VideoPlayerOverrides = {
  playButton?: {
    stylePreset?: MQ<string>;
  };
  seekBar?: {
    stylePreset?: MQ<string>;
    loadingProgressBar?: {
      stylePreset?: MQ<string>;
    };
    playProgressBar?: {
      stylePreset?: MQ<string>;
    };
    seekHandle?: {
      stylePreset?: MQ<string>;
      width?: MQ<string>;
    };
    currentDuration?: {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
    } & LogicalProps;
    seekPosition?: {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
    } & LogicalProps;
  };
  controlBar?: {
    stylePreset?: MQ<string>;
    typographyPreset?: MQ<string>;
    maxHeight?: MQ<string>;
    timeDisplay?: {
      recorded?: {
        typographyPreset?: MQ<string>;
        stylePreset?: MQ<string>;
        currentTime?: {
          spaceInline?: MQ<string>;
        };
        divider?: {
          spaceInline?: MQ<string>;
        };
      };
    };
    volumeControl?: {
      volumeBar?: {
        stylePreset?: MQ<string>;
      };
      volumeLevel?: {
        stylePreset?: MQ<string>;
      };
    };
    iconToggle?: {
      stylePreset?: MQ<string>;
    };
  };
  dockText?: {
    stylePreset?: MQ<string>;
    title?: {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
      spaceStack?: MQ<string>;
    };
    description?: {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
    };
  } & LogicalProps;
  miniCardOverlay?: {
    stylePreset?: MQ<string>;
    countdown?: {
      typographyPreset?: MQ<string>;
      stylePreset?: MQ<string>;
      spaceStack?: MQ<string>;
    };
    name?: {
      typographyPreset?: MQ<string>;
      stylePreset?: MQ<string>;
    };
    closeButton?: LogicalProps;
  } & LogicalProps;
} & LogicalProps;

export interface VideoPlayerProps {
  config: BrightCoveBasicAttributes & BrightCoveOtherAttributes;
  overrides?: VideoPlayerOverrides;
}
