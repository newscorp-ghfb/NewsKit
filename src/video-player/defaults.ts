export default {
  videoPlayer: {
    playButton: {
      stylePreset: 'videoPlayerPlayButton',
    },
    seekBar: {
      stylePreset: 'videoPlayerSeekBarContainer',
      loadingProgressBar: {
        stylePreset: 'videoPlayerLoadProgressBar',
      },
      playProgressBar: {
        stylePreset: 'videoPlayerPlayProgressBar',
      },
      seekHandle: {
        stylePreset: 'videoPlayerSeekHandle',
        width: 'borderWidth020',
      },
      currentDuration: {
        stylePreset: 'videoPlayerCurrentDuration',
        paddingInline: 'space020',
        paddingBlock: 'space020',
        typographyPreset: 'utilityMeta010',
      },
      seekPosition: {
        stylePreset: 'videoPlayerSeekPosition',
        paddingInline: 'space020',
        paddingBlock: 'space020',
        typographyPreset: 'utilityMeta010',
      },
    },
    controlBar: {
      stylePreset: 'videoPlayerControlBar',
      typographyPreset: {
        xs: 'utilityMeta010',
        md: 'utilityMeta020',
      },
      maxHeight: {
        xs: 'sizing070',
        md: 'sizing080',
      },
      timeDisplay: {
        typographyPreset: 'utilityMeta010',
        stylePreset: 'inkInverse',
        currentTime: {
          spaceInline: 'space010',
        },
        divider: {
          spaceInline: 'space010',
        },
      },
      volumeControl: {
        volumeBar: {
          stylePreset: 'videoPlayerVolumeBar',
        },
        volumeLevel: {
          stylePreset: 'videoPlayerVolumeLevel',
        },
      },
      iconToggle: {
        stylePreset: 'inkInverse',
      },
    },
    dockText: {
      stylePreset: 'videoPlayerDockText',
      paddingInline: {
        xs: 'space040',
        md: 'space050',
      },
      paddingBlock: {
        xs: 'space040',
        md: 'space050',
      },
      title: {
        stylePreset: 'videoPlayerDockTextTitle',
        typographyPreset: {
          xs: 'utilityHeading010',
          md: 'utilityHeading040',
        },
        spaceStack: {
          xs: 'space020',
          md: 'space030',
        },
      },
      description: {
        stylePreset: 'inkInverse',
        typographyPreset: {
          xs: 'utilityBody010',
          md: 'utilityBody030',
        },
      },
    },
    miniCardOverlay: {
      stylePreset: 'videoPlayerMiniCardOverlay',
      paddingInline: 'space030',
      paddingBlock: 'space020',
      countdown: {
        typographyPreset: 'utilityMeta010',
        stylePreset: 'inkInverse',
        spaceStack: 'space010',
      },
      name: {
        typographyPreset: 'utilityMeta010',
        stylePreset: 'inkInverse',
      },
      closeButton: {
        paddingInline: 'space010',
        paddingBlock: 'space010',
      },
    },
  },
};
