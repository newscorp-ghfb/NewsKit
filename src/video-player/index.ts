export * from './video-player';
export * from './types';
