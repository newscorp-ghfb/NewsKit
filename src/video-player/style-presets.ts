import {StylePreset} from '../theme/types';

export default {
  videoPlayerPlayButton: {
    base: {
      backgroundColor: '{{colors.transparent}}',
      borderStyle: 'solid',
      borderColor: '{{colors.white}}',
      borderWidth: '{{borders.borderWidth020}}',
      borderRadius: '{{borders.borderRadiusCircle}}',
      color: '{{colors.white}}',
    },
  },
  videoPlayerSeekBarContainer: {
    base: {
      backgroundColor: '{{colors.blackTint020}}',
    },
  },
  videoPlayerLoadProgressBar: {
    base: {
      backgroundColor: '{{colors.whiteTint050}}',
    },
  },
  videoPlayerPlayProgressBar: {
    base: {
      backgroundColor: '{{colors.interactivePrimary030}}',
    },
  },
  videoPlayerSeekHandle: {
    base: {
      backgroundColor: '{{colors.interactivePrimary030}}',
    },
  },
  videoPlayerCurrentDuration: {
    base: {
      backgroundColor: '{{colors.inkInverse}}',
      borderStyle: 'solid',
      borderColor: '{{colors.inkInverse}}',
      borderWidth: '{{borders.borderWidth010}}',
      borderRadius: '{{borders.borderRadiusRounded020}}',
      color: '{{colors.inkContrast}}',
    },
  },
  videoPlayerSeekPosition: {
    base: {
      backgroundColor: '{{colors.inkContrast}}',
      borderStyle: 'solid',
      borderColor: '{{colors.inkContrast}}',
      borderWidth: '{{borders.borderWidth010}}',
      borderRadius: '{{borders.borderRadiusRounded020}}',
      color: '{{colors.inkInverse}}',
    },
  },
  videoPlayerControlBar: {
    base: {
      backgroundColor: '{{colors.blackTint050}}',
    },
  },
  videoPlayerVolumeBar: {
    base: {
      backgroundColor: '{{colors.inkInverse}}',
    },
  },
  videoPlayerVolumeLevel: {
    base: {
      backgroundColor: '{{colors.interactivePrimary030}}',
    },
  },
  videoPlayerDockText: {
    base: {
      backgroundImage: '{{overlays.overlayGradientInverseVertical}}',
    },
  },
  videoPlayerDockTextTitle: {
    base: {
      color: '{{colors.white}}',
    },
  },
  videoPlayerMiniCardOverlay: {
    base: {
      backgroundColor: '{{colors.neutral090}}',
    },
  },
} as Record<string, StylePreset>;
