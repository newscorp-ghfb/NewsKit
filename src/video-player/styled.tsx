import {logicalProps} from '../utils/logical-properties';
import {
  getResponsiveBorder,
  getResponsiveSize,
  getResponsiveSpace,
  getStylePreset,
  getTypographyPreset,
  styled,
} from '../utils/style';
import {VideoPlayerProps} from './types';

export const VideoPlayerWrapper = styled.div<
  Pick<VideoPlayerProps, 'overrides'>
>`
  ${logicalProps()}
  //play button
  .video-js .vjs-big-play-button {
    ${getStylePreset('videoPlayer.playButton', 'playButton')}
  }

  //seekbar - container - collapsed
  .video-js .vjs-progress-holder {
    width: 100%;
    ${getStylePreset('videoPlayer.seekBar', 'seekBar')};
    )}
  }

  // seekbar - loading progress bar
  .video-js .vjs-load-progress {
    ${getStylePreset(
      'videoPlayer.seekBar.loadingProgressBar',
      'seekBar.loadingProgressBar',
    )};
    div {
      background: none;
    }
  }

  // seekbar - play progress bar
  .video-js .vjs-play-progress {
    ${getStylePreset(
      'videoPlayer.seekBar.playProgressBar',
      'seekBar.playProgressBar',
    )}
    .vjs-time-tooltip {
      ${getStylePreset(
        'videoPlayer.seekBar.currentDuration',
        'seekBar.currentDuration',
      )}
    }
  }
  //seekbar - currentDuration & seekPosition typographyPreset & spaceInset
  .video-js .vjs-time-tooltip {
    ${getTypographyPreset(
      'videoPlayer.seekBar.currentDuration',
      'seekBar.currentDuration',
    )}
    ${getResponsiveSpace(
      'padding',
      'videoPlayer.seekBar.currentDuration',
      'seekBar.currentDuration',
      'spaceInset',
    )}
  }

  //seekbar - seekhandle
  .video-js .vjs-progress-control .vjs-mouse-display {
    ${getStylePreset('videoPlayer.seekBar.seekHandle', 'seekBar.seekHandle')}
    ${getResponsiveBorder(
      borderSize => ({
        width: borderSize,
      }),
      'videoPlayer.seekBar.seekHandle',
      'seekBar.seekHandle',
      'width',
    )}
    .vjs-time-tooltip {
      ${getStylePreset(
        'videoPlayer.seekBar.seekPosition',
        'seekBar.seekPosition',
      )}
    }
  }
  //control-bar
  .video-js .vjs-control-bar {
    ${getStylePreset('videoPlayer.controlBar', 'controlBar')};
    ${getTypographyPreset('videoPlayer.controlBar', 'controlBar')};
    ${getResponsiveSize(
      'maxHeight',
      'videoPlayer.controlBar',
      'controlBar',
      'maxHeight',
    )}
  }

  //controlBar - timeDisplay - recorded
  .video-js .vjs-time-control {
    align-items: center;
    ${getStylePreset(
      'videoPlayer.controlBar.timeDisplay.stylePreset',
      'controlBar.timeDisplay.stylePreset',
    )}
    ${getTypographyPreset(
      'videoPlayer.controlBar.timeDisplay.typographyPreset',
      'controlBar.timeDisplay.typographyPreset',
    )}
  }
  //controlBar - timeDisplay - recorded - currentTime
  .video-js .vjs-current-time {
    ${getResponsiveSpace(
      'marginRight',
      'videoPlayer.controlBar.timeDisplay.currentTime',
      'controlBar.timeDisplay.currentTime',
      'spaceInline',
    )}
  }
  //controlBar - timeDisplay - recorded - divider
  .video-js .vjs-time-divider {
    ${getResponsiveSpace(
      'marginRight',
      'videoPlayer.controlBar.timeDisplay.divider',
      'controlBar.timeDisplay.divider',
      'spaceInline',
    )}
  }

  //control bar - volume container
  .video-js .vjs-volume-panel-horizontal {
    display: flex;
    align-items: center;
  }

  //control bar - volume bar
  .video-js .vjs-volume-bar {
    ${getStylePreset(
      'videoPlayer.controlBar.volumeControl.volumeBar',
      'controlBar.volumeControl.volumeBar',
    )}
  }

  //control bar - volume level
  .video-js .vjs-volume-level {
    ${getStylePreset(
      'videoPlayer.controlBar.volumeControl.volumeLevel',
      'controlBar.volumeControl.volumeLevel',
    )}
  }

  //control bar - Icon toggles
  .video-js .vjs-control {
    ${getStylePreset(
      'videoPlayer.controlBar.iconToggle',
      'controlBar.iconToggle',
    )}
  }

  // Dock text - text wrapper
  .video-js .vjs-dock-text {
    background: none;
    ${getStylePreset('videoPlayer.dockText', 'dockText')};
    ${getResponsiveSpace(
      'padding',
      'videoPlayer.dockText',
      'dockText',
      'spaceInset',
    )}
  }

  // Dock text - title
  .video-js .vjs-dock-title {
    ${getStylePreset('videoPlayer.dockText.title', 'dockText.title')};
    ${getTypographyPreset('videoPlayer.dockText.title', 'dockText.title')};
    ${getResponsiveSpace(
      'marginBottom',
      'videoPlayer.dockText.title',
      'dockText.title',
      'spaceStack',
    )}
  }
  // Dock text - description
  .video-js .vjs-dock-description {
    ${getStylePreset(
      'videoPlayer.dockText.description',
      'dockText.description',
    )};
    ${getTypographyPreset(
      'videoPlayer.dockText.description',
      'dockText.description',
    )};
  }

  //miniOverlayCard - container

  .video-js .vjs-next-overlay-banner {
    ${getStylePreset('videoPlayer.miniCardOverlay', 'miniCardOverlay')};
    ${getResponsiveSpace(
      'padding',
      'videoPlayer.miniCardOverlay',
      'miniCardOverlay',
      'spaceInset',
    )};
  }

  //miniOverlayCard - countdown

  .video-js .vjs-next-overlay-countdown {
    ${getTypographyPreset(
      'videoPlayer.miniCardOverlay.countdown',
      'miniCardOverlay.countdown',
    )};
    ${getStylePreset(
      'videoPlayer.miniCardOverlay.countdown',
      'miniCardOverlay.countdown',
    )};
    ${getResponsiveSpace(
      'marginBottom',
      'videoPlayer.miniCardOverlay.countdown',
      'miniCardOverlay.countdown',
      'spaceStack',
    )}
  }

  //miniOverlayCard - name

  .video-js .vjs-next-overlay-name {
    ${getTypographyPreset(
      'videoPlayer.miniCardOverlay.name',
      'miniCardOverlay.name',
    )};
    ${getStylePreset(
      'videoPlayer.miniCardOverlay.name',
      'miniCardOverlay.name',
    )};
  }

  //miniOverlayCard - closeButton
  .video-js .vjs-next-overlay {
    .vjs-close-button {
      ${getResponsiveSpace(
        'padding',
        'videoPlayer.miniCardOverlay.closeButton',
        'miniCardOverlay.closeButton',
        'spaceInset',
      )}
    }
  }
`;
