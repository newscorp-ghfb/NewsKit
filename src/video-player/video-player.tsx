import React, {useEffect} from 'react';
import {withOwnTheme} from '../utils/with-own-theme';
import defaults from './defaults';
import stylePresets from './style-presets';
import {VideoPlayerWrapper} from './styled';
import {VideoPlayerProps} from './types';

const getBrightcoveUrl = (accountID: string, playerID: string) =>
  `https://players.brightcove.net/${accountID}/${playerID}_default/index.min.js`;

const ThemelessVideoPlayer = React.forwardRef<
  HTMLVideoElement,
  VideoPlayerProps
>(({config, overrides}, ref) => {
  const {
    'data-account': accountID,
    'data-player': playerID,
    'data-video-id': videoID,
    'data-playlist-id': playListID,
  } = config;
  const id =
    playListID !== undefined && videoID === undefined
      ? `bc-video-${playListID}`
      : `bc-video-${videoID}`;

  useEffect(() => {
    const src = getBrightcoveUrl(accountID, playerID);
    const script = document.createElement('script');
    script.type = 'text/javascript';
    script.src = src;
    document.head.appendChild(script);
    // To be ignored in coverage as jest does not execute the script tags or load scripts
    /* istanbul ignore next */
    script.onload = () => {
      /* istanbul ignore else */
      if (window.videojs) {
        window.videojs.getPlayer(id).ready(function () {
          // @ts-ignore
          const player = this;
          /* istanbul ignore else */
          if (ref && 'current' in ref) {
            // eslint-disable-next-line no-param-reassign
            ref.current = player;
          }
        });
      }
    };
  }, [accountID, playerID, id, ref]);

  return (
    <VideoPlayerWrapper overrides={overrides}>
      <video // eslint-disable-line jsx-a11y/media-has-caption
        id={id}
        className="video-js vjs-fluid"
        data-testid="bc-video-player"
        {...config}
      />
    </VideoPlayerWrapper>
  );
});

export const VideoPlayer = withOwnTheme(ThemelessVideoPlayer)({
  defaults,
  stylePresets,
});
