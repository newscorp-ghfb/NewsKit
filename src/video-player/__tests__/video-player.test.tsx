import {renderWithTheme} from '../../test/test-utils';
import {VideoPlayer} from '../video-player';
import {VideoPlayerProps} from '../types';
import {
  DEFATULT_VIDEO_PLAYER_CONFIG,
  PLAYLISTS_VIDEO_PLAYER_CONFIG,
} from './config';
import {createTheme} from '../../theme';

const DefaultVideoPlayerProps: VideoPlayerProps = {
  config: DEFATULT_VIDEO_PLAYER_CONFIG,
};

const PlaylistVideoPlayerProps: VideoPlayerProps = {
  config: PLAYLISTS_VIDEO_PLAYER_CONFIG,
};

const myCustomTheme = createTheme({
  name: 'video-player-theme',
  overrides: {
    stylePresets: {
      customPlaybutton: {
        base: {
          borderColor: 'red',
          borderStyle: 'solid',
          borderWidth: '{{borders.borderWidth010}}',
          borderRadius: '{{borders.borderRadiusCircle}}',
          backgroundColor: '{{colors.amber050}}',
          color: '{{colors.red050}}',
        },
      },
    },
  },
});
describe('Video Player', () => {
  test('default player should render properly with data attributes', () => {
    const {getByTestId} = renderWithTheme(VideoPlayer, DefaultVideoPlayerProps);
    expect(getByTestId('bc-video-player')).toHaveAttribute(
      'data-account',
      DEFATULT_VIDEO_PLAYER_CONFIG['data-account'],
    );
    expect(getByTestId('bc-video-player')).toHaveAttribute(
      'data-video-id',
      DEFATULT_VIDEO_PLAYER_CONFIG['data-video-id'],
    );
    expect(getByTestId('bc-video-player')).toHaveAttribute(
      'data-player',
      DEFATULT_VIDEO_PLAYER_CONFIG['data-player'],
    );
    expect(getByTestId('bc-video-player').getAttribute('id')).toContain(
      DEFATULT_VIDEO_PLAYER_CONFIG['data-video-id'],
    );
    expect(getByTestId('bc-video-player')).toMatchSnapshot();
  });

  test('playlist player should render properly with data attributes', () => {
    const {getByTestId} = renderWithTheme(
      VideoPlayer,
      PlaylistVideoPlayerProps,
    );
    expect(getByTestId('bc-video-player')).toHaveAttribute(
      'data-account',
      PLAYLISTS_VIDEO_PLAYER_CONFIG['data-account'],
    );
    expect(getByTestId('bc-video-player')).toHaveAttribute(
      'data-playlist-id',
      PLAYLISTS_VIDEO_PLAYER_CONFIG['data-playlist-id'],
    );
    expect(getByTestId('bc-video-player')).toHaveAttribute(
      'data-player',
      PLAYLISTS_VIDEO_PLAYER_CONFIG['data-player'],
    );
    expect(getByTestId('bc-video-player').getAttribute('id')).toContain(
      PLAYLISTS_VIDEO_PLAYER_CONFIG['data-playlist-id'],
    );
    expect(getByTestId('bc-video-player')).toMatchSnapshot();
  });

  test(`renders video player with overriden play button style`, () => {
    const {asFragment} = renderWithTheme(
      VideoPlayer,
      {
        config: DEFATULT_VIDEO_PLAYER_CONFIG,
        overrides: {
          playButton: {
            stylePreset: 'customPlaybutton',
          },
        },
      },
      myCustomTheme,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test(`renders video player with logical props`, () => {
    const {asFragment} = renderWithTheme(
      VideoPlayer,
      {
        config: DEFATULT_VIDEO_PLAYER_CONFIG,
        overrides: {
          paddingInline: 'space020',
          paddingBlock: 'space040',
          marginBlock: 'space060',
          marginInline: 'space080',
        },
      },
      myCustomTheme,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
