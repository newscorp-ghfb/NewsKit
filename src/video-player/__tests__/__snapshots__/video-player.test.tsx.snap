// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Player default player should render properly with data attributes 1`] = `
<video
  class="video-js vjs-fluid"
  controls=""
  data-account="1752604059001"
  data-embed="default"
  data-player="default"
  data-testid="bc-video-player"
  data-video-id="5290445795001"
  id="bc-video-5290445795001"
/>
`;

exports[`Video Player playlist player should render properly with data attributes 1`] = `
<video
  class="video-js vjs-fluid"
  controls=""
  data-account="1752604059001"
  data-embed="default"
  data-player="S13cJdUBz"
  data-playlist-id="5686354139001"
  data-testid="bc-video-player"
  id="bc-video-5686354139001"
/>
`;

exports[`Video Player renders video player with overriden play button style 1`] = `
<DocumentFragment>
  .emotion-0 .video-js .vjs-big-play-button {
  border-color: red;
  border-style: solid;
  border-width: 1px;
  border-radius: 50%;
  background-color: #FF9900;
  color: #FF362E;
}

.emotion-0 .video-js .vjs-progress-holder {
  width: 100%;
  background-color: rgba(10,10,10,0.2);
}

<div
    class="emotion-0"
  >
    <video
      class="video-js vjs-fluid"
      controls=""
      data-account="1752604059001"
      data-embed="default"
      data-player="default"
      data-testid="bc-video-player"
      data-video-id="5290445795001"
      id="bc-video-5290445795001"
    />
  </div>
</DocumentFragment>
`;
